### PR TITLE
Longer client session identifiers

### PR DESCRIPTION
**Pull request by marcoconti83**
# Reason for this pull request
Adopting the new Cryptobox API for client sessions identifiers

# Testing
I'm not testing session migration (old identifiers to new identifiers) here. I tested it at the wire-ios-cryptobox level, and I plan to make an integration test at the wire-ios-sync-engine level when adopting the new API.

### CODE REVIEW

File: Source/Model/Message/ZMClientMessage+Encryption.swift [hunk 1]
```diff
@@ -229,9 +231,10 @@ extension ZMGenericMessage {
                         }
                     }
                     
-                    guard let encryptedData = try? sessionDirectory.encrypt(self.data(), recipientClientId: clientRemoteIdentifier) else {
+                    guard let encryptedData = try? sessionDirectory.encrypt(self.data(), for: clientRemoteIdentifier) else {
                         return nil
                     }
+                    print(clientRemoteIdentifier)
```
Comment (typfel) [inline Source/Model/Message/ZMClientMessage+Encryption.swift]: Delete?

Comment (marcoconti83) [inline Source/Model/Message/ZMClientMessage+Encryption.swift]: Yup

**Comment by mikeger** [comment]
Just a clarification, would there be a migration step from the existing identifiers to the new ones?

**Comment by marcoconti83** [comment]
@mikeger yes, there need to be one (done in wire-ios-sync-engine) or you will lose all sessions

**Comment by marcoconti83** [comment]
@typfel fixed

**Pull request merged**